### ISSUE TITLE
[$500] LHN - "No activity yet message showed in LHN rooms with conv history after coming back online

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#30392]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.91-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Create a new applause.expensifail account on staging.new.expensify.com
2. Disable the internet connection in the device
3. Tap on the green plus button (FAB) and create a new Workspace
4. Navigate to LHN
5. Enter any of the new rooms and send some messages
6. Go back online

## Expected Result:
Last send message into workspace rooms should be visible in LHN when come back online

## Actual Result:
"No activity yet" message showed in LHN under workspace rooms

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

 

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

 

https://github.com/Expensify/App/assets/78819774/63af195a-14d1-436e-9092-00ed120fd731

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014dee0189b20922e7</li>
        <li>Upwork Job ID: 1717286481120063488</li>
        <li>Last Price Increase: 2023-11-01</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @youssef-lr</details>

**Comment by alexpensify** [comment]
@s77rt - do you agree with this feedback from @bernhardoj?

**Comment by s77rt** [comment]
@alexpensify Yes! That makes sense to me. I have asked if that change was intended [here](https://github.com/Expensify/App/pull/29568#discussion_r1373755918). cc @youssef-lr

**Comment by youssef-lr** [comment]
Yeah this looks like a regression from my PR. I added this change for the case of split bills that contain a receipt, the other participant receives a last message text from the backend which says "split $0.0" when the receipt is still being scanned, even though this last message is actually a whisper to the creator of the split. So I think proper way is to get it fixed in the backend and we can revert my change at line 306.

**Comment by s77rt** [comment]
Should this be internal in that case?

**Comment by s77rt** [comment]
Not overdue. Maybe internal? ^

**Comment by alexpensify** [comment]
@youssef-lr can you check if we should take this internal and edit the tags here? Thanks!

**Comment by youssef-lr** [comment]
Going to take this internal and assign it to myself

**Comment by alexpensify** [comment]
Sounds good @youssef-lr, I'm going to assign you the primary role in this GH.

**Comment by youssef-lr** [comment]
I'm planning to work on this this week.

**Comment by youssef-lr** [comment]
Planning to get to this tomorrow.

**Comment by youssef-lr** [comment]
I sadly didn't get around to working on this and I'm going OOO tomorrow. I think for now we can revert this [change](https://github.com/Expensify/App/issues/30392#issuecomment-1780469569) which should fix this issue, we can work on fixing the backend issue afterwards.

**Comment by alexpensify** [comment]
Ok, we will wait for you to get back from OOO.

**Comment by alexpensify** [comment]
Still waiting on @youssef-lr here, no update.

**Comment by alexpensify** [comment]
@youssef-lr are you able to pick this one back up?

**Comment by youssef-lr** [comment]
@alexpensify fixing this one in the backend turned out to be a bit complicated, I'm afraid fixing the query might significantly slow it down, I'm going to double check in eng-chat

**Comment by alexpensify** [comment]
Thanks for the update!

**Comment by youssef-lr** [comment]
I'm exploring an alternative solution, will report back here next week!

**Comment by alexpensify** [comment]
Thank you @youssef-lr! 

**Heads up, I will be offline from Friday, December 22, to Thursday, January 4, 2024. I will not be actively watching over this GitHub during that period. If anything urgent is needed here, please ask for help in the #expensify-open-source Slack Room-- thanks!**

**Comment by youssef-lr** [comment]
No update yet, I'm still on it. I should have an update by EOW.

**Comment by youssef-lr** [comment]
It seems like the change we were planning to make is already in main, and still does not solve the issue. This seems like an issue with Onyx updates. I'm gonna keep investigating, but I think it's a low priority right now as the reproduction steps may not happen much in real world scenarios.

**Comment by youssef-lr** [comment]
No update yet.

**Comment by youssef-lr** [comment]
I haven't managed to get to this yet. I'm focusing on higher priority issues.

**Comment by youssef-lr** [comment]
No update yet, still a low priority.

**Comment by youssef-lr** [comment]
Same.

**Comment by youssef-lr** [comment]
No update yet.

**Comment by youssef-lr** [comment]
Same.

**Comment by youssef-lr** [comment]
No update yet

**Comment by youssef-lr** [comment]
Same.

**Comment by youssef-lr** [comment]
No update yet.

**Comment by youssef-lr** [comment]
No update yet, still a low priority.

**Comment by alexpensify** [comment]
## Heads up, I will be offline until Tuesday, May 28, 2024, and will not actively watch over this GitHub during that period.

If anything urgent is needed here, please ask for help in the #expensify-open-source Slack Room-- thanks!

**Comment by youssef-lr** [comment]
No update yet.

**Comment by youssef-lr** [comment]
Same.

**Comment by youssef-lr** [comment]
No update, probably worth retesting this it might have been resolved.

**Comment by youssef-lr** [comment]
Couldn't reproduce this anymore, closing!
